### PR TITLE
68904 Refactor covid19 vacine e2e test

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
The PR refactors the Covid19 vaccine appointment workflow e2e test.

## Related issue(s)

department-of-veterans-affairs/va.gov-team#68904

## Testing done
e2e testing.

## Screenshots
N/A

## What areas of the site does it impact?
VAOS

## Acceptance criteria
All test should pass

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/components/FormButtons.jsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ export default function FormButtons({
           onClick={onSubmit}
           disabled={disabled}
           className="usa-button usa-button-primary"
+          aria-label={nextButtonText || 'Continue'}
```
Comment (vbahinwillit) [inline src/applications/vaos/components/FormButtons.jsx:34]: Fixes 508 error.

File: src/applications/vaos/covid-19-vaccine/components/ReviewPage.jsx [hunk 2]
```diff
@@ -171,6 +171,7 @@ export default function ReviewPage({ changeCrumb }) {
                     name={facilityDetails.name}
                     facility={facilityDetails}
                     showDirectionsLink
+                    level={3}
```
Comment (vbahinwillit) [inline src/applications/vaos/covid-19-vaccine/components/ReviewPage.jsx:174]: Fixes 508 error

File: src/applications/vaos/tests/e2e/page-objects/AppointmentList/AppointmentListPageObject.js [hunk 3]
```diff
@@ -11,25 +30,6 @@ export class AppointmentListPageObject extends PageObject {
...
-    return this;
-  }
-
-  validate() {
-    // Wait for appointments to load
-    cy.wait(['@v2:get:appointments']);
-
-    cy.get('h2')
-      .should('be.visible')
-      .and('contain', 'Appointments');
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/e2e/page-objects/AppointmentList/AppointmentListPageObject.js:25]: This query was causes Cypress to crash in CI build. So it was refactored as 'assertAppointmentList'.

**Review by powellkerry — APPROVED** [review]

File: src/applications/vaos/components/FormButtons.jsx
Comment (simiadebowale) [inline src/applications/vaos/components/FormButtons.jsx:1]: can we add prop validation for this file please

File: src/applications/vaos/tests/e2e/fixtures/MockFacility.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export class MockFacility {
```
Comment (simiadebowale) [inline src/applications/vaos/tests/e2e/fixtures/MockFacility.js:1]: There's a new Mocks Facility created in `src/applications/vaos/tests/mocks/unit-test-helpers.js`. Maybe we could create a universal helper file that are reusable for both e2e and unit tests?

File: src/applications/vaos/components/FormButtons.jsx
Comment (vbahinwillit) [reply src/applications/vaos/components/FormButtons.jsx:1]: Sure.

**Pull request merged**